### PR TITLE
Added AI Chatbot

### LINKED ISSUE #88: Feat: Adding AI chatbot
### Describe the feature

An AI chatbot on Nacto Care website can provide 24/7 customer support, offer healthcare information. 
The chatbot can ask preliminary questions to understand the user’s needs and direct them to the appropriate healthcare professional.
The chatbot can provide users with information about different types of care, insurance coverage, and other services.

### Add ScreenShots

Screenshot 2024-05-21 013906

### Record

- [X] I agree to follow this project's Code of Conduct
- [X] I'm a GSSOC'24 contributor
- [X] I want to work on this issue

### PR DESCRIPTION
**Pull request by antriksh-9**
## Related Issue
Closes #88.

## Description
An AI chatbot on Nacto Care website can provide 24/7 customer support, offer healthcare information.
The chatbot can ask preliminary questions to understand the user’s needs and direct them to the appropriate healthcare professional.
The chatbot can provide users with information about different types of care, insurance coverage, and other services.

## Type of PR

- [ ] Bug fix
- [x] Feature enhancement
- [ ] Documentation update
- [ ] Other (specify): _______________

## Screenshots / videos (if applicable)

https://github.com/Nactore-Org/Nacto-Care/assets/132576984/49aeb9a9-2e91-4936-86a9-3891edb4b1cc

## Checklist:
- [x] I have performed a self-review of my code
- [x] I have read and followed the Contribution Guidelines.
- [x] I have tested the changes thoroughly before submitting this pull request.
- [x] I have provided relevant issue numbers, screenshots, and videos after making the changes.
- [ ] I have commented my code, particularly in hard-to-understand areas.

### CODE REVIEW

**Comment by Harshil-Jani** [comment]
@antriksh-9 UI wise looks great. I would like to understand where the data is coming in from ? Is it ChatGPT replying ? Or How it works overall ?

**Comment by antriksh-9** [comment]
> @antriksh-9 UI wise looks great. I would like to understand where the data is coming in from ? Is it ChatGPT replying ? Or How it works overall ?

Actually for creating bots through botpress we have to make the conversation flow between user and bot, like the greetings you see or the options that the bot produces for user to choose from, those are manually generated. These conversation flow are manually generated so that user get engaged with only the website related stuff and not outside that.
Now it's not only that the user get to only choose from the options, after the end of a conversation flow if a user asks a random question then also the bot can answer. For this we have to implement knowledge base, which of course I have added.

**Comment by antriksh-9** [comment]
Also these type of chatbots are not conversational AI, these are a specific goal-oriented, less flexible...
Here's a related article for reference - [blog](https://botpress.com/blog/difference-between-bot-conversational-ai)

**Comment by Harshil-Jani** [comment]
I see. So at the moment, its on your botpress account right ? In future if we need to edit something, it should be done by you only ? Since it's a no-code-tool on CDN.

I have a work-around for this. You can share the `.bpz` file by exporting the configurations you made on botpress and I can self host using nactore email and send the link back here.

**Comment by antriksh-9** [comment]
> I see. So at the moment, its on your botpress account right ? In future if we need to edit something, it should be done by you only ? Since it's a no-code-tool on CDN.
> 
> I have a work-around for this. You can share the `.bpz` file by exporting the configurations you made on botpress and I can self host using nactore email and send the link back here.

It says .bpz link is not supported.

**Comment by Harshil-Jani** [comment]
Weird ? Works for me. Can you send a screenshot where ? 

image

When I tried to export it downloads a .bpz file and then I can import it back on the studio and publish it.

**Comment by antriksh-9** [comment]
> Weird ? Works for me. Can you send a screenshot where ?
> 
> image
> 
> When I tried to export, it downloads a `.bpz` file and then I can import it back on the studio and publish it.

Yeah this works for me also, I was saying that I can't share the file here, as `.bpz` file type is not supported in github markdown.

**Comment by Harshil-Jani** [comment]
Oh my bad.  Can you mail it at nactoreorganization@gmail.com ?

**Comment by antriksh-9** [comment]
> Oh my bad. Can you mail it at [nactoreorganization@gmail.com](mailto:nactoreorganization@gmail.com) ?

Sent.

**Comment by antriksh-9** [comment]
Hey @Harshil-Jani, the file that I sent you it is working fine, right?

**Comment by Harshil-Jani** [comment]
@antriksh-9 Yes the knowledge base wass imported but the styles were missing. Probably we would need the Stylesheet configs. Have asked for them in mail itself. Can you please help with it ?

**Comment by antriksh-9** [comment]
> @antriksh-9 Yes the knowledge base wass imported but the styles were missing. Probably we would need the Stylesheet configs. Have asked for them in mail itself. Can you please help with it ?

The default style settings can be changed from the configuration sections (Integrations -> Configure Webchat) of the bot. 
I have selected `Dawn` theme and `green` widget color for the bot, also I have changed the avatar so that it matches the website theme.
I think the `.bpz` file only contains the conversation flow and knowledge base but not the style settings, so these need to be configured manually. And there is no stylesheet configs as the styles can be directly changed.

Screenshot 2024-05-25 133116

**Comment by Harshil-Jani** [comment]
@antriksh-9 Yeah. I see. I have explored this section earlier but I  thought that the themes you have set up are from the https://styler.botpress.app/ and it has a config option to share the stylesheet URL separately that goes right in the configuration.

Anyways, Here is the link to config https://mediafiles.botpress.cloud/b2905264-a15e-41ab-83e9-7db49dbaabc3/webchat/v2/config.js

**Comment by antriksh-9** [comment]
> @antriksh-9 Yeah. I see. I have explored this section earlier but I thought that the themes you have set up are from the https://styler.botpress.app/ and it has a config option to share the stylesheet URL separately that goes right in the configuration.
> 
> Anyways, Here is the link to config https://mediafiles.botpress.cloud/b2905264-a15e-41ab-83e9-7db49dbaabc3/webchat/v2/config.js

So, what work you want me to do now?

**Comment by Harshil-Jani** [comment]
Fairly simple, Now you need to change the URL in code from your's URL (which has config from your account) to the URL which I sent above.

File: src/components/Layout.jsx [hunk 1]
```diff
@@ -2,13 +2,18 @@ import React from "react";
...
 
 function Layout() {
   return (
     <div className="App">
       <Header />
       <Outlet />
       <Footer />
+      <Helmet>
+        <script src="https://cdn.botpress.cloud/webchat/v2/inject.js" defer async></script>
+        <script src="https://mediafiles.botpress.cloud/7b050b54-3832-4502-b938-a6a018f8f015/webchat/v2/config.js" defer async></script>
```
Comment (Harshil-Jani) [inline src/components/Layout.jsx:15]: Over here @antriksh-9 in case of confusion.

**Comment by antriksh-9** [comment]
> Fairly simple, Now you need to change the URL in code from your's URL (which has config from your account) to the URL which I sent above.

I have made the changes.

**Review by Harshil-Jani — APPROVED** [review]

**Pull request merged**